### PR TITLE
Fix #5565: Revert the {pos,run}-separate-compilation changes

### LINKED ISSUE #5565: Please revert the {pos,run}-separate-compilation changes
I believe splitting out separate compilation into new directories was done so that we would not get pickler failures when we compile all of pos/run. This imposes a tax on everyone using the test framework because things will be misclassified. In particular everyone coming from Scala-2, since they don't have a structure like this.

Would it not be far simpler if the pickling tests skipped all tests that were directories? That's maybe a two-line change in the test framework vs a constant hassle for its users.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/test/dotty/tools/vulpix/ParallelTesting.scala [hunk 1]
```diff
@@ -1242,9 +1242,13 @@ trait ParallelTesting extends RunnerOrchestration { self =>
 
     val (dirs, files) = compilationTargets(sourceDir, fileFilter)
 
+    val doSequentialSeparateCompilation = !flags.options.contains("-Ytest-pickler")
+    def compileDir(dir: JFile): Boolean = {
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Can you add a comment explaining why some directories need to be filtered out?

Comment (allanrenucci) [inline compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Rename to `ignoreDir`? This name suggests to me that the method is compiling the directory

Comment (nicolasstucki) [reply compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Changed

Comment (nicolasstucki) [reply compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Added

File: compiler/test/dotty/tools/dotc/CompilationTests.scala [hunk 2]
```diff
@@ -90,10 +90,9 @@ class CompilationTests extends ParallelTesting {
     compileFilesInDir("tests/new", defaultOptions) +
     compileFilesInDir("tests/pos-scala2", scala2Mode) +
     compileFilesInDir("tests/pos", defaultOptions) +
-    compileFilesInDir("tests/pos-separate-compilation", defaultOptions) +
     compileFilesInDir("tests/pos-deep-subtype", allowDeepSubtypes) +
     compileFilesInDir("tests/pos-kind-polymorphism", defaultOptions and "-Ykind-polymorphism") +
-    compileDir("tests/pos-separate-compilation/i1137-1", defaultOptions) +
+    compileDir("tests/pos/i1137-1", defaultOptions) +
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/test/dotty/tools/dotc/CompilationTests.scala]: This test is already covered by https://github.com/lampepfl/dotty/pull/5571/files/90b45b5023b9290593f5e342708b269646b07a30#diff-6bca83394d6f42a29e212ad536fda738R92

File: compiler/test/dotty/tools/vulpix/ParallelTesting.scala [hunk 3]
```diff
@@ -1242,9 +1242,13 @@ trait ParallelTesting extends RunnerOrchestration { self =>
...
+      doSequentialSeparateCompilation || !dir.listFiles().exists(file => file.getName.endsWith(".java") || file.getName.endsWith("_2.scala"))
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Not if the java file depends on Scala defined classes.

File: compiler/test/dotty/tools/dotc/CompilationTests.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/test/dotty/tools/dotc/CompilationTests.scala]: Removed

File: compiler/test/dotty/tools/vulpix/ParallelTesting.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/test/dotty/tools/vulpix/ParallelTesting.scala]: Now only `_2.java` classes are disallowed. The failing tests are blacklisted.

**Pull request merged**